File: sdks/dotnet/src/Mobiscroll.Connect/Exceptions/ValidationException.cs
using System.Text.Json;

namespace Mobiscroll.Connect.Exceptions;

public sealed class ValidationException : MobiscrollConnectException
{
    /// <summary>Raw validation detail payload returned by the API, if any.</summary>
    public JsonElement? Details { get; }

    public ValidationException(string message, JsonElement? details = null)
        : base(message, "VALIDATION_ERROR")
    {
        Details = details;
    }
}

File: sdks/java/connect-sdk/src/main/java/com/mobiscroll/connect/exceptions/ValidationException.java
package com.mobiscroll.connect.exceptions;

import com.fasterxml.jackson.databind.JsonNode;

/** Thrown on HTTP 400 / 422 — request payload was rejected by the server. */
public class ValidationException extends MobiscrollConnectException {

    private static final long serialVersionUID = 1L;

    private final transient JsonNode details;

    public ValidationException(String message, JsonNode details) {
        super(message);
        this.details = details;
    }

    /** Raw {@code details} field from the server error body, or {@code null} if not present. */
    public JsonNode getDetails() {
        return details;
    }
}
